django_irods/views.py

```diff
@@ -270,10 +270,13 @@ def download(request, path, use_async=True, use_reverse_proxy=True,
             return JsonResponse(task_dict)
     else:  # regular file download
         # if fetching main metadata files, then these need to be refreshed.
-        if path.endswith("resourcemap.xml") or path.endswith('resourcemetadata.xml'):
+        if path in [f"{res_id}/data/resourcemap.xml", f"{res_id}/data/resourcemetadata.xml",
+                    f"{res_id}/manifest-md5.txt", f"{res_id}/tagmanifest-md5.txt", f"{res_id}/readme.txt",
+                    f"{res_id}/bagit.txt"]:
             metadata_dirty = res.getAVU("metadata_dirty")
             if metadata_dirty is None or metadata_dirty:
-                create_bag_files(res)  # sets metadata_dirty to False
+                create_bag_metadata_files(res)  # sets metadata_dirty to False
+                create_bagit_files_by_irods(res, res.get_irods_storage())
```

Comment:
Wouldn't it be better to make ```create_bagit_files_by_irods``` method to take only one resource parameter and get irods storage by calling ```res.get_irods_storage()``` inside the method rather than passing it as a parameter?

Reply:
I did that because get_irods_storage() creates a new instance each time.  This allows me to avoid creating the instance again in the task here https://github.com/hydroshare/hydroshare/pull/4102/files#diff-182216ce9ee2cc3e1f410a911873b12dc126f5e025f7b81f8d5d1ecdfe234b0bR358

Reply:
OK. That makes sense.